packages/process/process_error.pony

```diff
@@ -0,0 +1,54 @@
+class val ProcessError
+  let error_type: ProcessErrorType
+  let message: (String | None)
+
+  new val create(error_type': ProcessErrorType,
+    message': (String | None) = None)
+  =>
+    error_type = error_type'
+    message = message'
+
+  fun string(): String =>
+    match message
+    | let m: String => error_type.string() + ": " + m
```

Comment:
Having the 'string()` function return a `String val` might be confusing, as `Stringable` in stdlib is defined as returning a `String iso^`. I would also make the `ProcessErrorType` primitives return a `String iso^` to be in line with Stringable as well.

May I suggest to use sth like this in this line here?

```pony
let err_type_name = error_type.string()
let msg = 
  recover iso 
    let tmp = String(err_type_name.size() + 2 + m.size())
    tmp.>append(consume err_type_name)
    .>append(": ")
    .>append(m)
end
```

Reply:
I was thinking that the returned string should not be mutable.

Reply:
I think String iso^ is preferable so it lines up with Stringable.

Reply:
Being Stringable was my intent when I suggested the idea.

Reply:
OK, updated.